### PR TITLE
Never use instance variables in partials

### PR DESCRIPTION
**Pull request by killthekitten**
It is a bad practice to use instance variables in partials for several reasons. Some more of them can be found here http://rails-bestpractices.com/posts/2010/07/24/replace-instance-variable-with-local-variable/

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -833,6 +833,12 @@ when you need to retrieve a single record by some attributes.
   Mitigate code duplication by using partial templates and layouts.
 <sup>[[link](#partials)]</sup>
 
+* <a name="no-instance-variables-in-partials"></a>
+  Never use instance variables in partials, pass a local variable to `render` instead. Partial may be used
```
Comment (bbatsov) [inline README.md]: I'd change `never` to `avoid`, as plenty of people are doing so. Some concrete examples here would be helpful.

Comment (killthekitten) [inline README.md]: 👍 done

File: README.md [hunk 2]
```diff
@@ -833,6 +833,26 @@ when you need to retrieve a single record by some attributes.
...
+  <%= render 'course_description' %>
+  # good
+  <%= render 'course_description', course: @course %>
+  
+  # app/views/courses/_course_description.html.erb
+  # bad
+  <%= @course.description %>
+  # good
+  <%= course.description %>
+  ```
```
Comment (bquorning) [inline README.md]: I think the meaning would be clearer if the entire bad example was in one listing, and the good one as well:
```erb
# bad
# app/views/courses/show.html.erb
<%= render 'course_description' %>
# app/views/courses/_course_description.html.erb
<%= @course.description %>

# good
# app/views/courses/show.html.erb
<%= render 'course_description', course: @course %>
# app/views/courses/_course_description.html.erb
<%= course.description %>
```

File: README.md [hunk 3]
```diff
@@ -833,6 +833,26 @@ when you need to retrieve a single record by some attributes.
   Mitigate code duplication by using partial templates and layouts.
 <sup>[[link](#partials)]</sup>
 
+* <a name="no-instance-variables-in-partials"></a>
+  Avoid using instance variables in partials, pass a local variable to `render` instead. Partial may be used
```
Comment (andyw8) [inline README.md]: Can I suggest some small grammar changes:

**The** partial may be used in a different controller or action, where the variable can have a different name or **is absent**. In these cases **an** undefined instance variable will not raise an exception whereas **a** local variable will.

File: README.md (hunk 2 above)
Comment (killthekitten) [reply README.md]: They say it's better late than never! Updated.

File: README.md (hunk 3 above)
Comment (killthekitten) [reply README.md]: Yes, this way it reads so much better, thanks!

**Comment by killthekitten** [comment]
I guess this one can be safely squashed back into master now?

**Comment by bbatsov** [comment]
Indeed! Thanks!

**Pull request merged**